src/02-common-principles.md

```diff
@@ -153,50 +176,55 @@ The solutions will change from case to case and publicly available datasets will
 be reviewed to include common data types in the future releases of the BIDS
 specification.
 
-## File name structure
-
-A file name consists of a chain of *entities*, a *suffix* and an *extension*.
-Two prominent examples of entities are `subject` and `session`.
-
-For a data file that was collected in a given `session` from a given
-`subject`, the file name MUST begin with the string `sub-<label>_ses-<label>`.
-If the `session` level is omitted in the folder structure, the file name MUST begin
-with the string `sub-<label>`, without `ses-<label>`.
-
-Note that `sub-<label>` corresponds to the `subject` entity because it has
-the `sub-` "key" and`<label>` "value", where `<label>` would in a real data file
-correspond to a unique identifier of that subject, such as `01`.
-The same holds for the `session` entity with its `ses-` key and its `<label>`
-value.
-
-The extra session layer (at least one `/ses-<label>` subfolder) SHOULD
-be added for all subjects if at least one subject in the dataset has more than
-one session.
-If a `/ses-<label>` subfolder is included as part of the directory hierarchy,
-then the same [`ses-<label>`](./99-appendices/09-entities.md#ses)
-entity MUST also be included as part of the file names themselves.
-Acquisition time of session can
-be defined in the [sessions file](03-modality-agnostic-files.md#sessions-file).
-
-A chain of entities, followed by a suffix, connected by underscores (`_`)
-produces a human readable file name, such as `sub-01_task-rest_eeg.edf`.
-It is evident from the file name alone that the file contains resting state
-data from subject `01`.
-The suffix `eeg` and the extension `.edf` depend on the imaging modality and
-the data format and indicate further details of the file's contents.
-
-Entities within a file name MUST be unique.
-For example, the following file name is not valid because it uses the `acq`
-entity twice:
-`sub-01_acq-laser_acq-uneven_electrodes.tsv`
-
-In cases where entities duplicate metadata,
-the presence of an entity should not be used as a replacement for
+## Filesystem structure
+
+Data for each subject are placed in sub-directories named "`sub-<label>`",
+where string "`<label>`" is substituted with the unique identification
+label of each subject. Additional information on each participant MAY be
+provided in a [participants file](03-modality-agnostic-files.md#participants-file)
+in the root directory of the dataset.
+
+If data for the subject were acquired across multiple sessions, then within
+the subject directory resides sub-directories named "`ses-<label>`",
+where string "`<label>`" is substituted with a unique identification
+label for each session. In datasets where at least one subject has more than
+one session, this additional sub-directory later SHOULD be added for all
+subjects in the dataset. Additional information on each session MAY be
```

Comment:
I know this recommendation was in the spec prior to this PR, but I don't really get the point: What if subjects miss some sessions ... or if it's a between subject design with different subjects being part of different sessions?

See also the recommendation below:

> A data type directory SHOULD NOT be defined if
there are no files to be placed in that directory.

We don't have to discuss it here, but if someone has a quick point on this, I am happy to hear it.

Reply:
My suspicion is that it was intended to make things easier for App developers. If a hypothetical App is hard-wired to loop over multiple sessions within a subject and process each individually, but without adequate logic to detect the absence of a session structure in any particular subject, passing such a dataset could break the App. Personally I'm happy to have them break; but I'm a known software sadist so that might not count for much.